WordPress/Classes/Utility/iAds/SearchAdsAttribution.swift

```diff
@@ -0,0 +1,124 @@
+/// Implementation of Search Ads attribution details
+/// More info: https://searchads.apple.com/help/measure-results/#attribution-api
+
+import Foundation
+import iAd
+import AutomatticTracks
+
+@objc final class SearchAdsAttribution: NSObject {
+
+    /// Keep the instance alive
+    ///
+    private static var lifeToken: SearchAdsAttribution?
+
+    private static let userDefaultsSentKey = "search_ads_attribution_details_sent"
+    private static let userDefaultsLimitedAdTrackingKey = "search_ads_limited_tracking"
+
+    /// Is ad tracking limited?
+    /// If the user has limited ad tracking, and this API won't return data
+    ///
+    private var isTrackingLimited: Bool {
+        get {
+            return UserDefaults.standard.bool(forKey: SearchAdsAttribution.userDefaultsLimitedAdTrackingKey)
+        }
+        set {
+            UserDefaults.standard.set(newValue, forKey: SearchAdsAttribution.userDefaultsLimitedAdTrackingKey)
+        }
+    }
+
+    /// Has the attribution details been sent already?
+    /// Once the attribution details are sent, it won't change; so it is not necesary to send it again
+    ///
+    private var isAttributionDetailsSent: Bool {
+        get {
+            return UserDefaults.standard.bool(forKey: SearchAdsAttribution.userDefaultsSentKey)
+        }
+        set {
+            UserDefaults.standard.set(newValue, forKey: SearchAdsAttribution.userDefaultsSentKey)
+        }
+    }
+
+    #if (arch(i386) || arch(x86_64)) && os(iOS)
+    private let isSimulator = true
+    #else
+    private let isSimulator = false
+    #endif
+
+
+    override init() {
+        super.init()
+        SearchAdsAttribution.lifeToken = self
+    }
+
+    @objc func requestDetails() {
+        guard
+            isSimulator == false, // Requests from simulator will always fail
+            isTrackingLimited == false,
+            isAttributionDetailsSent == false
+        else {
+            finish()
+            return
+        }
+
+        requestAttributionDetails()
+    }
+
+    private func requestAttributionDetails() {
+        ADClient.shared().requestAttributionDetails { [weak self] (attributionDetails, error) in
+            DispatchQueue.main.async {
+                if let error = error as NSError? {
+                    self?.didReceiveError(error)
+                } else {
+                    self?.didReceiveAttributionDetails(attributionDetails)
+                }
+            }
+        }
+    }
+
+    private func didReceiveAttributionDetails(_ details: [String: NSObject]?) {
+        defer {
+            finish()
+        }
+        guard let details = details?["Version3.1"] as? [String: Any] else { return }
```

Comment:
`return` should go in it's own line. [Details here!](https://github.com/wordpress-mobile/swift-style-guide#braces)

Reply:
(Plus) `Version3.1` should probably be a constant.